base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -925,25 +927,28 @@ pub fn calculate_mmr_roots<T: BlockchainBackend>(db: &T, block: &Block) -> Resul
     let deleted = deleted.deleted;
     let mut kernel_mmr = MerkleMountainRange::<HashDigest, _>::new(kernels);
     let mut output_mmr = MutableMmr::<HashDigest, _>::new(outputs, deleted)?;
-    let mut proof_mmr = MerkleMountainRange::<HashDigest, _>::new(range_proofs);
+    let mut witness_mmr = MerkleMountainRange::<HashDigest, _>::new(range_proofs);
+    let mut input_mmr = MutableMmr::<HashDigest, _>::new(Vec::new(), Bitmap::create())?;
 
     for kernel in body.kernels().iter() {
         kernel_mmr.push(kernel.hash())?;
     }
 
     for output in body.outputs().iter() {
         output_mmr.push(output.hash())?;
-        proof_mmr.push(output.proof().hash())?;
+        witness_mmr.push(output.witness_hash())?;
     }
 
     for input in body.inputs().iter() {
+        let input_hash = input.hash();
```

Comment:
This does not technically include the entire input. We are only hashing the following fields here:  
```
features
commitment
script
```
We need all the fields. 

My proposal would be to rename the function `input.hash()` to `input.txo_hash()` or something along those lines. This hash would then match the hash of output.hash(). 
We then make `input.hash()` the hash of the entire `TransactionInput`

Reply:
Ok that makes sense - perhaps `input.output_hash()` and we assert that it matches the input hash

Reply:
Changes made